### ISSUE TITLE
How to change the content of the web ? 

### ISSUE THREAD

**Issue by e212156822000** [AKSHAYUBHAT/DeepVideoAnalytics#68]
## Context of the issue
I knew it sounds stupid .......
I made some changes in `templates/base.html` , but there is still no change on the web.
I'm pretty sure I've changed the content.

## Expected result
The website should changed when the time I changed the the code.

## Already tried
1. killed docker's containers
2. `docker-compose up --force-recreate`
3. `docker-compose down -v`
I've known how to build django with docker-compose, but I've never countered this issue before.

**Comment by AKSHAYUBHAT** [comment]
1. In the current docker-compose.yml there is "git pull" which pulls the latest version of the repo from Github before starting the container. You need to make sure that it is removed, otherwise it might just overwrite it. 

2. Another issue might be that if you are editing after server has started the base template might have been cached in memory of the process. You can try to force django processes to reload, which does not happens automatically if the debug mode is disabled.

**Comment by e212156822000** [comment]
Thanks for your help. :sob: :sob:
I've already deleted `git pull` from docker/Dockerfile from the beginning.
I think the problem is cache. I tried some new ways.

### clear django's cache
I run `docker-compose run dva python manage.py shell`

`>>> from django.core.cache import cache`
`>>> cache.clear()`

[@reference--django-clear-cache](https://github.com/rdegges/django-clear-cache)

### set debug mode to false
Also I set the DEBUG mode to False in `dva/settings.py`.
```python
# SECURITY WARNING: don't run with debug turned on in production!
if 'DISABLE_DEBUG' in os.environ or HEROKU_DEPLOY:
    DEBUG = False
else:
    DEBUG = False # change
```
### delete django's sessions
I run `docker-compose run dva python manage.py  clearsessions`

### delete some suspicious code
I'm not sure if `RUN git clone https://github.com/akshayubhat/DeepVideoAnalytics /root/DVA` in `docker/Dockerfile` affect this situation. I deleted it. It did not show any error.

### result
Still cahce is sticking with me. :sweat_drops: :sweat_drops:

**Comment by AKSHAYUBHAT** [comment]
If you use docker run then it will create a shell inside a NEW container and not an existing one. If your goal is to persist changes. You need to put either the entire DVA directory or templates directory inside a volume and then use `docker exec` to enter  running and NOT `docker run`.

See this on how enter into a shell inside a running container.

https://github.com/AKSHAYUBHAT/DeepVideoAnalytics/blob/master/docker/shell.sh

**Comment by e212156822000** [comment]
Thanks for your helping , you saved my day !!
When I get into the shell , I found the code not changed. 
Why not just replaced `RUN git clone https://github.com/akshayubhat/DeepVideoAnalytics /root/DVA` with `ADD ../ /root/DVA` in Dockerfile? Perhaps someone will encounter this issue?

## How to fix it
 
1. `docker/Dockerfile` 
```
FROM akshayubhat/dva_cpu_base
MAINTAINER Akshay Bhat <akshayubhat@gmail.com>
RUN youtube-dl -U
RUN mkdir /root/DVA
WORKDIR "/root/DVA"
RUN pip install --upgrade cffi
RUN pip install --no-deps keras
RUN pip install -r requirements.txt
RUN apt-get install -y libhdf5-dev
RUN pip install --no-deps h5py
WORKDIR "/root/DVA"
RUN cd dvalib/data && wget https://www.dropbox.com/s/jytpgw8et09ede9/facenet.pb && cd ../..
RUN cd dvalib/data && wget https://www.dropbox.com/s/3yzonc9nzo9xanv/vgg.pb && cd ../..
RUN cd dvalib/data/2016_08 && wget https://www.dropbox.com/s/umo0xtljm9an90x/model.ckpt && wget https://www.dropbox.com/s/f93axdtlb3ltj40/model.ckpt.meta && cd ../../..
RUN cd dvalib/yolo/model_data && wget https://www.dropbox.com/s/zbff2rkoejx5k5r/yolo.h5
WORKDIR "/root/DVA/repos/lopq/python"
RUN python setup.py install
WORKDIR "/root/DVA"
VOLUME ["/root/DVA/dva/media"]
ADD ../ /root/DVA
EXPOSE 80
```
if you use GPU mode then change `docker/Dockerfile.gpu`.

2. `docker/docker-compose.yml` , add code below the volume tag.
(partial)
```yml
dva:
   volumes:
       - dvadata:/root/DVA/dva/media
       - ..:/root/DVA
```

**Comment by AKSHAYUBHAT** [comment]
The `RUN git clone.... ` command runs when a new container image is created and now when `docker-compose up` is executed. Since creating a container is much more time consuming, it typically far easier to do a git pull or update source file some other way. Currently I build a new image only when a new library is added which needs to be installed/compiled.
